Packages/vcs/Lib/VTKAnimate.py

```diff
@@ -41,48 +138,85 @@ def __init__(self,vcs_self):
         import atexit
         atexit.register(self.close)
 
-    def draw_frame(self, frame_num = None):
+    def extract_renderers(self):
+        if self.cleared:
+            return
+        self.cleared = True
+        be = self.vcs_self.backend
+        if be.renWin is None: #Nothing to clear
+            return
+        renderers = be.renWin.GetRenderers()
+        renderers.InitTraversal()
+        ren = renderers.GetNextItem()
+
+        be.hideGUI()
+        self.create_thread.hidden_window.SetNumberOfLayers(be.renWin.GetNumberOfLayers())
+        while ren is not None:
+            if not ren.GetLayer() == 0:
+                be.renWin.RemoveRenderer(ren)
+                self.create_thread.hidden_window.AddRenderer(ren)
+            else:
+                self.create_thread.bg_ren.SetBackground(*ren.GetBackground())
+            ren = renderers.GetNextItem()
+        be.showGUI()
+
+        self.create_thread.hidden_window.SetSize(*be.renWin.GetSize())
+
+    def reclaim_renderers(self):
+        if not self.cleared:
+            return
+        self.cleared = False
+
+        be = self.vcs_self.backend
+
+        if be.renWin is None: #Nothing to clear
+            return
+
+        be.hideGUI()
+        renderers = be.renWin.GetRenderers()
+        renderers.InitTraversal()
+        ren = renderers.GetNextItem()
```

Comment:
I have to think more about it but this could lead to problems if the renderer knows about the context from the previous render window.